ppx/codegen.ml

```diff
@@ -132,9 +145,9 @@ let find_body_factory ~loc input_nested_tuple_pattern output_expression
     connection_function_expr input_nested_tuple_expression =
   [%expr
     let f result =
-      Result.map
-        ~f:(fun [%p input_nested_tuple_pattern] -> [%e output_expression])
-        result
+      match result with
+      | Ok [%p input_nested_tuple_pattern] -> [%e output_expression]
```

Comment:
Small bug: these two lines should be `| Ok ... -> Ok ...` and `| Error e -> Error e`

Reply:
(I fixed before merging)